lib/url.js

```diff
@@ -495,7 +495,7 @@ Url.prototype.resolveObject = function(relative) {
     }
 
     result.protocol = relative.protocol;
-    if (!relative.host && !hostlessProtocol[relative.protocol]) {
+    if (!relative.host && !/file:?/.test(relative.protocol) && !hostlessProtocol[relative.protocol]) {
```

Comment:
Shouldn't the regex match the whole string?

Style: long line, please wrap at 80 columns (one clause per line.)